app/src/main/java/com/winlator/contents/ContentsManager.java

```diff
@@ -292,6 +345,120 @@ public static void cleanTmpDir(Context context) {
         file.mkdirs();
     }
 
+    /**
+     * Normalizes Wine/Proton library structure to ensure consistent layout.
+     * All Wine/Proton installations should have lib/wine/ subdirectory structure
+     * regardless of what the profile.json specifies.
+     *
+     * Expected structure after normalization:
+     * - bin/ (executables)
+     * - lib/ (shared libraries)
+     *   └── wine/ (Wine DLLs by architecture)
+     *       ├── i386-windows/
+     *       ├── x86_64-windows/
+     *       └── aarch64-windows/ (for arm64ec)
+     */
+    private void normalizeWineLibraryStructure(File installPath, ContentProfile profile) {
```

Comment:
This ensures that Proton and Wine distributions are all extracted the same way regardless of any differences in their profile.json.

This way, none of the handlers such launchers need to worry. The end result is a fully compatible installation of Wine that is essentially identical to the already pre-built wine binaries.